packages/squiggle-lang/__tests__/Distributions/Symbolic_test.res

```diff
@@ -0,0 +1,177 @@
+open Jest
+open Expect
+
+let fnImage = (theFn, inps) => Js.Array.map(theFn, inps)
+
+let env: DistributionOperation.env = {
+  sampleCount: 100,
+  xyPointLength: 100,
+}
+
+let mkNormal = (mean, stdev) => GenericDist_Types.Symbolic(#Normal({mean: mean, stdev: stdev}))
+let {toFloat, toDist, toString, toError, fmap} = module(DistributionOperation.Output)
+let run = DistributionOperation.run(~env)
+let outputMap = fmap(~env)
+let toExtFloat: option<float> => float = E.O.toExt(
+  "Should be impossible to reach (This error is in test file)",
+)
+let toExtDist: option<GenericDist_Types.genericDist> => GenericDist_Types.genericDist = E.O.toExt(
+  "Should be impossible to reach (This error is in a test file)", 
+)
+let unpackFloat = x => x -> toFloat -> toExtFloat
+let unpackDist = y => y -> toDist -> toExtDist
+
+describe("(Symbolic) normalize", () => {
+  testAll("has no impact on normal distributions", list{-1e8, -16.0, -1e-2, 0.0, 1e-4, 32.0, 1e16}, mean => {
+    let theNormal = mkNormal(mean, 2.0)
+    let theNormalized = run(FromDist(ToDist(Normalize), theNormal))
+    theNormalized 
+    -> unpackDist
+    -> expect
+    -> toEqual(theNormal)
+  })
+})
+
+describe("(Symbolic) mean", () => {
+  testAll("of normal distributions", list{-1e8, -16.0, -1e-2, 0.0, 1e-4, 32.0, 1e16}, mean => {
+    run(FromDist(ToFloat(#Mean), mkNormal(mean, 4.0))) 
+    -> unpackFloat 
+    -> expect 
+    -> toBeCloseTo(mean)
+  })
+
+  Skip.test("of normal(0, -1) (it NaNs out)", () => {
+    run(FromDist(ToFloat(#Mean), mkNormal(1e1, -1e0)))
+    -> unpackFloat
+    -> expect
+    -> ExpectJs.toBeFalsy
+  })
+
+  test("of normal(0, 1e-8) (it doesn't freak out at tiny stdev)", () => {
+    run(FromDist(ToFloat(#Mean), mkNormal(0.0, 1e-8)))
+    -> unpackFloat
+    -> expect
+    -> toBeCloseTo(0.0)
+  })
+
+  testAll("of exponential distributions", list{1e-7, 2.0, 10.0, 100.0}, rate => {
+    let theMean = run(FromDist(ToFloat(#Mean), GenericDist_Types.Symbolic(#Exponential({rate: rate}))))
```

Comment:
I suggest writing a helper function
``getMean: symbolic -> result<float, Error>``
so you don't have to type this full thing everwhere

Reply:
I thought about that but didn't want a reader of the test to have more layers of indirection when they're trying to decompose how stuff works.

Reply:
I think in this case it's repeated enough that it makes it less readable, but I don't feel incredibly strongly.